macros/graph/plotly3D.pl

```diff
@@ -144,21 +145,73 @@ =head1 FUNCTION TYPES
+          single value. This function type is useful to plot more complicated
+          functions, such as piecewise functions with if/then statements.
+          For example, this graphs the surface of the plane in the first
+          octant that passes through the points ($a,0,0), (0,$b,0), (0,0,$c):
+
+          ($a, $b, $c) = (5, 3, 7);
+          $graph->addSurface(
+              [
+                  "return ($b*u < $a*v ? 0.5*u : u - 0.5*$a/$b*v);",
+                  "return ($b*u > $a*v ? 0.5*v : v - 0.5*$b/$a*u);",
+                  "const x = ($b*u < $a*v ? 0.5*u : u - 0.5*$a/$b*v);"
+                . "const y = ($b*u > $a*v ? 0.5*v : v - 0.5*$b/$a*u);"
+                . "return $c - $c/$a*x - $c/$b*y;",
+              ],
+              [0, $a],
+              [0, $b],
+              funcType => 'js',
+          );
 
   perl    The functions are interpreted as Perl subroutines. The functions
           will be passed the appropriate number of inputs, and return a
-          single value.
+          single value. This uses the WeBWorK server to generate the points
+          for the plot, and can slow down the rendering of the problem. Using
+          the JavaScript methods are preferred for this reason. Here is an
+          example of plotting a sphere of radius $R.
+
+          $R = 5;
+          $graph->addSurface(
+              [
+                  sub { return $R*cos($_[0])*sin($_[1]); },
+                  sub { return $R*sin($_[0])*sin($_[1]); },
+                  sub { return $R*cos($_[1]); }
+              ],
+              [0, 2*pi],
+              [0, pi],
+              funcType => 'perl',
+          );
 
   data    The functions are interpreted as a nested array of data points to
-          be sent directly to plotly to plot. This is useful for static plots
-          in which the points do not need to be generated each time.
+          be sent directly to plotly to plot. The nested array needs to be
+          a string, since it is passed to JavaScript to plot. This array lists
+          all of the points which are used to create the surface. For example to
+          plot a surface with 9 points, use something like:
+
+          $graph->addSurface(
+              [
+                  "[[x1, x2, x3], [x4, x5, x6], [x7, x8, x9]]",
+                  "[[y1, y2, y3], [y4, y5, y6], [y7, y8, y9]]",
+                  "[[z1, z2, z3], [z4, z5, z6], [z7, z8, z9]]"
+              ],
+              [0,0],
+              [0,0],
+              funcType => 'data'
+          );
+
+          This plots a surfacing using the points (x1,y1,z1), (x2,y2,z2), ...,
+          and (x9,y9,z9). The addSurface method requires bounds, but they are not
+          used, so [0,0] needs to be included, but is ignored. Using the perl method
+          to first generate the arrays, then copying the result and using the data
+          method can be useful to speed up rendering of nonrandomized plots.
+
 
 =head1 Graph3D OPTIONS
 
```

Comment:
It would look a little better if the line below this (218) used monospace 
`C<$graph = Graph3D(options =E<gt> value)>`

Reply:
What tool do you use to render the POD, so I can see how it is formatted in HTML?